### PR TITLE
Add Log Summarizer

### PR DESCRIPTION
**Pull request by masonturner**
First pass at creating some useful functionality

### CODE REVIEW

File: infrastructure/logging/log_summarizer.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+int main(int argc, char* argv[]) {
+  const std::optional<std::string> log_path = parse_commandline_arguments(argc, argv);
+  Config log_config;
+  if (!log_path.has_value()) {
+    std::cout << "Please enter a log path" << std::endl;
+    return 0;
+  }
+
+  jet::LogReader log_reader(*log_path, {});
```
Comment (jpanikulam) [inline infrastructure/logging/log_summarizer.cc]: Consider making a function that takes a path to a log as an argument, and returns a summary

File: infrastructure/logging/log_summarizer.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  // Go through all the messages
+  for (std::size_t i = 0; i < channels.size(); ++i) {
+    jet::Message message = {};
+    if (log_reader.read_next_message(channels[i], message)) {
+      channel_stats[i].first_message_received = message.header.timestamp_ns;
+      channel_stats[i].num_messages_received++;
+    }
+    while(log_reader.read_next_message(channels[i], message)) {
+      channel_stats[i].last_message_received = message.header.timestamp_ns;
```
Comment (jpanikulam) [inline infrastructure/logging/log_summarizer.cc]: This is not a crime, but you should always feel a slight tinge of shame when doing this

Where “this” is getting the last element of a list by iterating through it and setting something equal to every element

File: infrastructure/logging/log_summarizer.cc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      channel_stats[i].num_messages_received++;
+    }
+    channel_stats[i].frequency = static_cast<double>(channel_stats[i].num_messages_received)/(std::chrono::duration_cast<std::chrono::seconds>(std::chrono::nanoseconds(channel_stats[i].last_message_received - channel_stats[i].first_message_received))).count();
```
Comment (jpanikulam) [inline infrastructure/logging/log_summarizer.cc]: `experiments/estimation/time_point.hh` has these operations

The ones I think you want are `to_time_point(const Timestamp&)`, and `estimation::to_seconds(const Duration&)`

In any case, you should factor your std::chrono operations into functions like `get_seconds()`

File: infrastructure/logging/log_summarizer.cc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  // Print out results;
+  const std::string reset("\033[0m");
+  const std::string red_bold("\033[31;1m");
+  std::cout << red_bold << std::setw(30) << "Channel" << std::setw(25) << "First Timestamp" << std::setw(25) << "Last Timestamp" << std::setw(15) << "Num Messages" << std::setw(15) << "Frequency (Hz)" << std::endl;
+  std::cout << reset;
+  for (std::size_t i = 0; i < channels.size(); ++i) {
+    std::cout << std::setw(30) << channels[i] << 
+                 std::setw(25) << std::fixed << std::setprecision(9) << channel_stats[i].first_message_received/static_cast<double>(1e9) << 
+                 std::setw(25) << std::fixed << std::setprecision(9) << channel_stats[i].last_message_received/static_cast<double>(1e9) <<
```
Comment (jpanikulam) [inline infrastructure/logging/log_summarizer.cc]: Good — this is usually more readable in the form of *differences* in units of seconds.

Ex: First/last message w.r.t first IMU message, or w.r.t first message received at all

File: infrastructure/logging/log_summarizer.cc (hunk 2 above)
Comment (masonturner) [reply infrastructure/logging/log_summarizer.cc]: let's talk about how I can fix this when you get back. adding a todo for now

**Pull request merged**